### DISCUSSION TITLE
umi4 如何动态修改base

### DISCUSSION THREAD

**Discussion by luanwulin** [umijs/umi#8475] [Q&A] [answered]
在umi3的时候可以通过打开`runtimeHistory `，在app.ts里面可以动态修改`window.routerBase = 'xxx'`

umi4没有了这个选项，应该如何实现呢？

**Comment by luanwulin** [comment, marked as answer]
看了下源码，，目前好像已经取消这种取值方式了。只有通过自定义plugins来解决了。
```js
import { IApi } from 'umi';

export default (api: IApi) => {
  // Runtime Plugin
  api.addRuntimePlugin(() => ['../plugin-routerBase/runtime.ts']);
  api.addRuntimePluginKey(() => ['routerBase']);

  api.onGenerateFiles(() => {
    api.writeTmpFile({
      path: '../plugin-routerBase/runtime.ts',
      content: `
export function modifyContextOpts(memo: any) {
  return  { ...memo, basename: window.routerBase };
};
`,
    });
  });
};

```

**Comment by twinkle77** [comment]
这种方式还是有点问题

<img width="383" alt="image" src="https://user-images.githubusercontent.com/17792166/202832328-b31c2b59-245e-4b5d-b80f-f94030d857f0.png">

访问 ```http://localhost:8888/``` 会默认打到  ```http://localhost:8888/ant-design-pro``` 路由
